### PR TITLE
Reduce size of html payload

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
The PR compresses and minifies the html returned from runner.
There is a significant reduction in request size.

### How to review 
Check that the html response from runner is compresses and minified.

### CODE REVIEW

**Comment by nealedj** [comment]
I'm sure it's negligible but it might be worth considering if this has much of an impact on the time it takes the application to process each request.

**Comment by iwootten** [comment]
Are you able to give an indication on the size of the reduction?

**Comment by jonnyshaw89** [comment]
@iwootten IIRC one page went from about 20kb to 4kb

File: app/setup.py [hunk 1]
```diff
@@ -178,6 +183,19 @@ def apply_caching(response):  # pylint: disable=unused-variable
 
         return response
 
+    @application.after_request
+    def response_minify(response):  # pylint: disable=unused-variable
```
Comment (MebinAbraham) [inline app/setup.py:186]: Worth adding a test to ensure this is working.
Have tested manually and looked good.

File: app/setup.py [hunk 2]
```diff
@@ -178,6 +183,19 @@ def apply_caching(response):  # pylint: disable=unused-variable
...
+        """
+        minify html response to decrease site traffic
```
Comment (MebinAbraham) [inline app/setup.py:188]:
```suggestion
        Minify HTML response to reduce the request size
```

File: app/setup.py (hunk 1 above)
Comment (jonnyshaw89) [reply app/setup.py:186]: What would the test include? checking that there are no spaces? is there much value there?

Comment (MebinAbraham) [reply app/setup.py:186]: We could check the `content-length` in the header request is smaller when the minify is enabled?. And `Content-Encoding: gzip` if that is available?.

Comment (bitdivision) [reply app/setup.py:186]: We would essentially be testing a configuration flag here?

Comment (MebinAbraham) [reply app/setup.py:186]: Which we already do?.

Comment (jonnyshaw89) [reply app/setup.py:186]: Also the current testing harness wont support binary responses like we would get with gzip enabled. to support this we would need to re-write the testing harness, which seems like quite an overhead

Comment (MebinAbraham) [reply app/setup.py:186]: Had a quick fiddle and in the testing, the content-length header is available which is smaller when the minify is set to true. But to explicitly check `gzip` maybe not.

**Review by MebinAbraham — APPROVED** [review]

**Review by bitdivision — APPROVED** [review]
Tested this with profiling turned on and got 0.003136 total time in compress over 42 requests. 0.00007466666 seconds per request.

Haven't tested minification with profiling

**Pull request merged**